### PR TITLE
OS2FORMS-371 Added easy access to  assigned maestro task by token

### PR DESCRIPTION
**Pull request by andriyun**
Followup for PR130

- New token ['webform_submission:'os2forms_forloeb_execute_task]
- Refactored assigned task fetching method
- Added ability to open os2forms_forloeb_execute_task URL without being logged in as Drupal User

### CODE REVIEW

**Review by rimi-itk — APPROVED** [review]

File: os2forms_forloeb.module [hunk 1]
```diff
@@ -293,10 +293,36 @@ function os2forms_forloeb_tokens($type, array $tokens, array $data, array $optio
...
 }
+
+/**
+ * Implements hook_entity_access().
+ * Allows requests with tokens to view the entity.
+ */
+function os2forms_forloeb_entity_access(\Drupal\Core\Entity\EntityInterface $entity, $operation, \Drupal\Core\Session\AccountInterface $account) {
+  if ($operation == 'update' && $entity instanceof WebformSubmission) {
+    $token = \Drupal::request()->query->get('os2forms-forloeb-ws-token');
+    if ($token && $token == $entity->getToken()) {
```
Comment (rimi-itk) [inline os2forms_forloeb.module:311]: For consistency, we should always use `===` for comparison.

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php [hunk 2]
```diff
@@ -151,6 +153,17 @@ public function getExecutableForm($modal, MaestroExecuteInteractive $parent) {
       $new_submission->bundle(), $taskUniqueSubmissionId, $sid
     );
 
-    return parent::getExecutableForm($modal, $parent);
+    $form = parent::getExecutableForm($modal, $parent);
+    // Catch os2forms-forloeb access token and pass it further.
+    if ($form instanceof RedirectResponse && $token = \Drupal::request()->query->get('os2forms-forloeb-ws-token')) {
+      // Check token to previous submission and update it to new one.
+      if ($token == $webform_submission->getToken()) {
```
Comment (rimi-itk) [inline src/Plugin/EngineTasks/MaestroWebformInheritTask.php:160]: Use `===` here.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. `===` should be used for comparisons rather than just `==`.

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php (hunk 2 above)
Comment (rimi-itk) [reply src/Plugin/EngineTasks/MaestroWebformInheritTask.php:160]: @andriyun, do you have local commits that haven't beed pushed?

Comment (andriyun) [reply src/Plugin/EngineTasks/MaestroWebformInheritTask.php]: @rimi-itk I just pushed fixes for these two changes you requested

File: os2forms_forloeb.module (hunk 1 above)
Comment (andriyun) [reply os2forms_forloeb.module]: Fixed

**Review by rimi-itk — APPROVED** [review]
Very good job.

**Comment by rimi-itk** [comment]
Hmm … I cannot merge either, @andriyun. @mBoegvald & @madsnorgaard, can you help us out?

**Comment by madsnorgaard** [comment]
I will squash and merge. Hope you are all well!

**Pull request merged**